src/main/java/seedu/duke/Task.java

```diff
@@ -0,0 +1,45 @@
+package seedu.duke;
+
+public abstract class Task {
+
+    protected String task;
+    protected Boolean isDone;
+    protected Character abbreviation;
+
+    /**
+     * Constructor for the Task Object
+     * @param task
+     * @throws Exception (Description of task cannot be empty)
+     */
+    public Task(String task) throws DukeException {
+        if (task == "") {
+            throw new DukeException("OOPS!!! The description of a task cannot be empty.");
+        }
+        this.task = task;
+        this.isDone = false;
+    }
+
+    public void mark() {
+        isDone = true;
+    }
+
+    public void unmark() {
+        isDone = false;
+    }
```

Comment:
Perhaps these two methods can be combined into one setter?

```suggestion
    public void mark(boolean isDone) {
        this.isDone = isDone;
    }
```